api/src/main/java/org/apache/gravitino/model/ModelChange.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.gravitino.model;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.gravitino.annotation.Evolving;
+
+/**
+ * A model change is a change to a model. It can be used to rename a model, update the comment of a
+ * model, set a property and value pair for a model, or remove a property from a model.
+ */
+@Evolving
+public interface ModelChange {
+  /**
+   * Create a ModelChange for renaming a model.
+   *
+   * @param newName The new model name.
+   * @return A ModelChange for the rename.
+   */
+  static ModelChange rename(String newName) {
+    return new ModelChange.RenameModel(newName);
+  }
+
+  /**
+   * Create a unsupported ModelChange for testing.
+   *
+   * @param message the error message.
+   * @return A ModelChange for the unsupported operation.
+   */
+  static ModelChange unSupport(String message) {
+    return new UnsupportModelChange(message);
+  }
```

Comment:
I think it is improper to add a `UnsupportModelChange` here only for testing. API module is exposed to the users, we should not add any unneeded things. Please remove it.